kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/KafkaConnectService.java

```diff
@@ -0,0 +1,228 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.retry.Retry;
+
+import java.util.Map;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class KafkaConnectService {
+    private static final int MAX_RETRIES = 5;
+
+    private final ClustersStorage clustersStorage;
+    private final WebClient webClient;
+
+    public Flux<String> getConnectors(String clusterName) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.get()
+                        .uri(cluster.getKafkaConnect() + "/connectors")
+                        .retrieve()
+                        .bodyToFlux(new ParameterizedTypeReference<String>() {
+                        })
+                        .retryWhen(
+                                Retry.onlyIf(e -> e.exception() instanceof WebClientResponseException.Conflict)
+                                        .retryMax(MAX_RETRIES))
+                        .doOnError(log::error)
+                )
+                .orElse(Flux.error(new NotFoundException("No such cluster")));
+    }
+
+    public Mono<Connector> createConnector(String clusterName, Mono<NewConnector> connector) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.post()
+                        .uri(cluster.getKafkaConnect() + "/connectors")
+                        .body(BodyInserters.fromPublisher(connector, NewConnector.class))
+                        .retrieve()
+                        .bodyToMono(Connector.class)
+                        .retryWhen(
+                                Retry.onlyIf(e -> e.exception() instanceof WebClientResponseException.Conflict)
+                                        .retryMax(MAX_RETRIES))
+                        .doOnError(log::error))
+                .orElse(Mono.error(new NotFoundException("No such cluster")));
+    }
+
+    public Mono<Connector> getConnector(String clusterName, String connectorName) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.get()
+                        .uri(cluster.getKafkaConnect() + "/connectors/{connectorName}", connectorName)
+                        .retrieve()
+                        .bodyToMono(Connector.class)
+                        .retryWhen(
+                                Retry.onlyIf(e -> e.exception() instanceof WebClientResponseException.Conflict)
+                                        .retryMax(MAX_RETRIES))
+                        .doOnError(log::error))
```

Comment:
What happens if retry logic exceeds the **MAX_RETRIES**?
Do we handle this case and what response do we return to clients?

Reply:
Added handling, thanks